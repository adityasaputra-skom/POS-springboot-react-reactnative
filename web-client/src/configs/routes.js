import ItemListPage from "../scenes/item";
import ItemPage from "../scenes/item/detail";
import UnitListPage from "../scenes/unit";
import UnitPage from "../scenes/unit/detail";
import TransactionListPage from "../scenes/transaction";
import YearSummmary from "../scenes/transaction/summary/YearSummary";
import MonthSummmary from "../scenes/transaction/summary/MonthSummary";
import DaySummmary from "../scenes/transaction/summary/DaySummary";
import TransactionPage from "../scenes/transaction/detail";
import StockListPage from "../scenes/stock";
import StockSummary from "../scenes/stock/summary/StockSummary";
import StockPage from "../scenes/stock/detail";
import Home from "../scenes/home";
import ErrorPage from "../scenes/error";

const routes = [
  {
    path: "/",
    component: Home
  },
  {
    path: "/items/add",
    component: ItemPage
  },
  {
    path: "/items/:id",
    component: ItemPage
  },
  {
    path: "/items",
    component: ItemListPage
  },
  {
    path: "/units/add",
    component: UnitPage
  },
  {
    path: "/units/:id",
    component: UnitPage
  },
  {
    path: "/units",
    component: UnitListPage
  },
  {
    path: "/transactions/year",
    component: YearSummmary
  },
  {
    path: "/transactions/month",
    component: MonthSummmary
  },
  {
    path: "/transactions/days",
    component: DaySummmary
  },
  {
    path: "/transactions/add",
    component: TransactionPage
  },
  {
    path: "/transactions/:id",
    component: TransactionPage
  },
  {
    path: "/transactions",
    component: TransactionListPage
  },
  {
    path: "/stocks/add",
    component: StockPage
  },
  {
    path: "/stocks/summary",
    component: StockSummary
  },
  {
    path: "/stocks/:id",
    component: StockPage
  },
  {
    path: "/stocks",
    component: StockListPage
  },
  {
    path: "*",
    component: ErrorPage,
    props: {
      code: 404
    }
  }
];

export default routes;
