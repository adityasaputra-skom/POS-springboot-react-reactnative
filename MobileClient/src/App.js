import 'react-native-gesture-handler';
import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {stackRoutes} from './configs/stackRoutes';
import {Root} from 'native-base';
import {Provider} from 'react-redux';
import store from './configs/store';
const Stack = createStackNavigator();
console.disableYellowBox = true;
export default function App() {
  return (
    <Provider store={store}>
      <Root>
        <NavigationContainer>
          <Stack.Navigator headerMode={'none'}>
            {stackRoutes.map((route, index) => (
              <Stack.Screen
                key={index}
                name={route.name}
                component={route.component}
              />
            ))}
          </Stack.Navigator>
        </NavigationContainer>
      </Root>
    </Provider>
  );
}
