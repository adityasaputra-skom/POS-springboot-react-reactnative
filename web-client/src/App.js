import React, { Component } from "react";
import "./App.css";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import routes from "./configs/routes";

class App extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <Router basename="/web-client">
        <Switch>
          {routes.map((route, index) => (
            <Route
              key={index}
              path={route.path}
              exact
              render={props => <route.component {...props} {...route.props} />}
            />
          ))}
        </Switch>
      </Router>
    );
  }
}

export default App;
