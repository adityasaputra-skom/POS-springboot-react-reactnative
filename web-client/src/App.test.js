import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import routes from "./configs/routes";
import toJson from "enzyme-to-json";

describe("App", () => {
  const wrapper = shallow(<App />);
  test("should ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("should ", () => {
    expect(wrapper.find("Route")).toHaveLength(routes.length);
  });
});
