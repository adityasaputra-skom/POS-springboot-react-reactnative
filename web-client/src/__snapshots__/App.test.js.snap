// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should  1`] = `
<BrowserRouter>
  <Switch>
    <Route
      exact={true}
      key="0"
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      key="1"
      path="/items"
      render={[Function]}
    />
    <Route
      exact={true}
      key="2"
      path="/units/add"
      render={[Function]}
    />
    <Route
      exact={true}
      key="3"
      path="/units/:id"
      render={[Function]}
    />
    <Route
      exact={true}
      key="4"
      path="/units"
      render={[Function]}
    />
    <Route
      exact={true}
      key="5"
      path="/transactions"
      render={[Function]}
    />
    <Route
      exact={true}
      key="6"
      path="/stocks"
      render={[Function]}
    />
    <Route
      exact={true}
      key="7"
      path="*"
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
